lib/extensions/attachable/active_record/base.rb

```diff
@@ -1,32 +1,137 @@
-    # This function should be declared in model, to it have attachments.
-    def has_many_attachments # rubocop:disable Style/PredicateName
+    # This method should be declared in model, to let it have attachments.
+    #
+    # @param [Hash] options
+    # @option options [Symbol|Array<Symbol>] :on The column which is associated with attachments,
+    #   the column type should be string or text.
+    #   This can be a symbol or an array of symbols.
+    #   An attribute named `column_name_attachment_references` will be defined, you can override it
+    #   to customise the way to retrieve the attachment_references for the specific column.
+    # @example Has many attachments on a column
+    #   has_many_attachments on: :description #=> description is associated with the attachments
+    #   of the model, updating description will result in attachments changing.
+    #   You can further implement `description_attachment_references_removed` reader in this case to
+    #   override the default method. The attachment_references ids returned by it will be removed.
+    def has_many_attachments(options = {}) # rubocop:disable Style/PredicateName
+      include HasManyAttachments
+
+      if options[:on]
+        self.attachable_columns = Array(options[:on])
+        before_save :update_attachment_references
+
+        HasManyAttachments.define_attachment_references_readers(attachable_columns)
+      end
+    end
+
+    def has_one_attachment # rubocop:disable Style/PredicateName
+      include HasOneAttachment
+    end
+  end
+
+  module HasManyAttachments
+    extend ActiveSupport::Concern
+
+    included do
+      class_attribute :attachable_columns
+      self.attachable_columns ||= []
+
       has_many :attachment_references, as: :attachable, class_name: "::#{AttachmentReference.name}",
                                        inverse_of: :attachable, dependent: :destroy, autosave: true
       # Attachment references can substitute attachments, so allow access using the `attachments`
       # identifier.
       alias_method :attachments, :attachment_references
+    end
+
+    ATTACHMENT_REMOVED_SUFFIX = '_attachment_references_removed'.freeze
+
+    def self.define_attachment_references_readers(attachable_columns)
+      attachable_columns.each do |column|
+        method_name = "#{column}#{ATTACHMENT_REMOVED_SUFFIX}"
+        next if method_defined?(method_name)
 
-      define_method(:files=) do |files|
-        files.each do |file|
-          attachment_references.build(file: file)
+        # Define a reader `#{column_name}_attachment_references_removed`, client code can override
+        # this method to implement their own logic.
+        # `column` and `column_was` are compared, only the differ attachment_references ids will be
+        # returned.
+        define_method(method_name) do
+          return [] unless send("#{column}_changed?")
+          ids_was = parse_attachment_reference_ids_from_content(send("#{column}_was"))
```

Comment:
Oh it's part of ActiveRecord::Dirty. I think good to add that into the comments - will update.